plugins/inputs/s7comm/s7comm.go

```diff
@@ -317,6 +317,20 @@ func handleFieldAddress(address string) (*gos7.S7DataItem, converterFunc, error)
 		if extra < 1 {
 			return nil, nil, fmt.Errorf("invalid extra parameter %d", extra)
 		}
+	case "X":
+		// We require an extra parameter
+		x := groups["extra"]
+		if x == "" {
+			return nil, nil, errors.New("extra parameter required")
+		}
+
+		bit, err = strconv.Atoi(x)
+		if err != nil {
+			return nil, nil, fmt.Errorf("invalid extra parameter: %w", err)
+		} else if bit < 0 || bit > 7 {
+			// Ensure bit address is valid
+			return nil, nil, fmt.Errorf("invalid extra parameter %d, bit address out of range", bit)
```

Comment:
What do you think of this?
```suggestion
			return nil, nil, fmt.Errorf("invalid extra parameter: bit address %d out of range", bit)
```